engine-types/src/types/wei.rs

```diff
@@ -188,6 +192,14 @@ impl Sub for Wei {
     }
 }
 
+impl Mul for Wei {
```

Comment:
I do not like this kind of implementation for newtypes because the units do not work properly. For example, if I have a new type representing meters (the unit for length) and there is this kind of implementation, then we would be able to write code where meters times meters gives you back meters (a measure of length) instead of meters^2 (a measure of area). Clearly this is incorrect.

It is ok to have implementations for `Add` and `Sub` because it does make sense to add together meters and get back meters. It is also ok to have multiplication by a pure scalar number (e.g. `u64`) because it does make sense to multiply a unit by a pure number.

I suggest if multiplication is needed then we have `impl Mul<U256> for Wei` instead.

Reply:
I got your point and agree with it. Good catch 👍🏻